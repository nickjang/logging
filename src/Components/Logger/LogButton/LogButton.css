.log-button {
  padding: 40px;
  margin: 12px auto 40px auto;
}

.log-button-status {
  margin: 28px auto 0px auto;
}