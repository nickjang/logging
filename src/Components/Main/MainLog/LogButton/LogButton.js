import React, { Component } from 'react';
import PropTypes from 'prop-types';

class LogButton extends Component {
  state = {
    log: {
      logging: false,
      start: null,
      stop: null
    },
    fetchError: {
      code: '',
      message: ''
    }
  }

  componentDidMount() {
    //fetch, logging project still?
  }

  startStopLog = () => {
    //if start, fetcherror/logging, fetch, setstate start // if never stop
    //if stop, fetcherror/logging, fetch, setstate stop, add log to MainLog state for main log list 

  }

  render() {
    return (
      <>
        <output className='form-status'>{this.state.fetchError || (this.state.log.logging && 'Logging...')}</output>
        <input 
          type='button' 
          value='start' 
          aria-label='Start a log.'
          onClick={(e) => this.startStopLog(e)}
        >Start/Stop
        </input>
      </>
    );
  }
}

LogButton.defaultProps = {
  addLog: () => {}
}

LogButton.propTypes = {
  addLog: PropTypes.func.isRequired
}

export default LogButton;