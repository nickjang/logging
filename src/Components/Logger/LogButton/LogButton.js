import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './LogButton.css';

class LogButton extends Component {
  state = {
    log: {
      logging: false,
      start: null,
      end: null
    },
    fetchError: ''
  }

  componentDidMount() {
    //fetch, logging project still?
  }

  startEndLog = () => {
    //if start, fetcherror/logging, fetch, setstate start // if never stop
    //if stop, fetcherror/logging, fetch, setstate stop, add log to MainLog state for main log list 

  }

  render() {
    return (
      <>
        <output className='form-status log-button-status'>{this.state.fetchError || (this.state.log.logging && 'Logging...')}</output>
        <input
          type='button'
          value={this.state.log.logging ? 'End' : 'Start'}
          aria-label='Start or end a log.'
          className='log-button'
          onClick={(e) => this.startEndLog(e)}/>
      </>
    );
  }
}

LogButton.defaultProps = {
  addLog: () => { }
}

LogButton.propTypes = {
  addLog: PropTypes.func.isRequired
}

export default LogButton;