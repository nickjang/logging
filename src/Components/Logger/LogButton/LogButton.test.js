import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LogButton from './LogButton';

describe(`LogButton component`, () => {
  it('renders LogButton by default', () => {
    const wrapper = shallow(<LogButton />);
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})