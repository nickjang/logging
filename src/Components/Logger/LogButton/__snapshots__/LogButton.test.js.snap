// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogButton component renders LogButton by default 1`] = `
<Fragment>
  <output
    className="form-status log-button-status"
  />
  <input
    aria-label="Start or end a log."
    className="
            log-button 
            log-button-start"
    disabled={true}
    onClick={[Function]}
    type="button"
    value="Start"
  />
</Fragment>
`;
